### PR TITLE
fix(django): django asgi support (when using debug mode)

### PR DESCRIPTION
**Pull request by mabdinur**
## Description
This PR mitigates the following issue: https://github.com/DataDog/dd-trace-py/issues/4305. Accessing ‘request.user’ is not supported in django async context.  

Support for ddtrace user tags in Django ASGI should be added in a follow up PR.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

**Review by juanjux — DISMISSED** [review]

File: releasenotes/notes/fix-cors-errors-0fa947a1a583ad0c.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    django: fix django asgi support when running ddtrace in debug mode
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/fix-cors-errors-0fa947a1a583ad0c.yaml:4]:
```suggestion
    django: avoid ``SynchronousOnlyOperation`` when failing to retrieve user information.
```

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]

**Comment by mabdinur** [comment]
@Mergifyio backport 1.6 1.5

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

File: ddtrace/contrib/django/utils.py [hunk 2]
```diff
@@ -292,6 +292,10 @@ def _after_request_tags(pin, span, request, response):
             # Note: getattr calls to user / user_is_authenticated may result in ImproperlyConfigured exceptions from
             # Django's get_user_model():
             # https://github.com/django/django/blob/a464ead29db8bf6a27a5291cad9eb3f0f3f0472b/django/contrib/auth/__init__.py
+            #
+            # FIXME: getattr calls to user fail in async contexts.
```
Comment (P403n1x87) [inline ddtrace/contrib/django/utils.py:296]: Would using `object.__getattribute__` solve this issue entirely?

Comment (brettlangdon) [reply ddtrace/contrib/django/utils.py:296]: I don't think so, it seems like the User attributes are delay fetched and fetched synchronously.

What we probably need to do is get an example app together for Django asgi/async that causes this same error and see how we should fetch this user data that is safe for async context.

**Pull request merged**